<launch>

  <node pkg="exploration" type="exploration_b" name="planner" output="screen">
    <param name="neighbourhood" value="3"/>
    <param name="base_frame" value="base_link"/>
    <param name="debug" value="False"/>
    <param name="radius" value="0.15"/>
    <remap from="/planner/occ_grid" to="/map"/>
    <remap from="/planner/goal" to="/move_base_simple/goal"/>
  </node>


  <node pkg="exploration" type="path_o_b" name="path_optimizer" output="screen">
      <param name="max_acceleration" value="0.3"/> 
      <param name="max_braking" value="0.1"/> 
      <param name="velocity" value="0.1"/> 
      <remap from="/path_optimizer/path" to="/planner/path"/>
  </node>

  <node pkg="exploration" type="path_f_b" name="path_follower" output="screen">
      <param name="Kx" value="1.0"/>
      <param name="Ky" value="-2.0"/>
      <param name="Ktheta" value="1.0"/>
      <param name="max_rot_speed" value="1.0"/>
      <param name="max_velocity" value="0.1"/>
      <param name="max_y_error" value="0.25"/>
      <param name="look_ahead" value="0.5"/>
      <param name="base_frame" value="base_link"/>
      <remap from="/path_follower/traj" to="/path_optimizer/trajectory"/>
      <!-- <remap from="/path_follower/twistCommand" to="/mux/autoCommand"/> -->
      <remap from="/path_follower/twistCommand" to="/obstacle_avoidance/command_velocity"/>
  </node>

  <node pkg="obstacle_avoidance" type="obstacle_avoidance" name="obstacle_avoidance" output="screen">
      launch-prefix="xterm -e gdb --args"
      <param name="base_frame" value="base_link" />
      <param name="max_range" value="5.0" />
      <param name="max_linear_velocity" value="1.0" />
      <param name="max_angular_velocity" value="1.0" />
      <param name="max_linear_accel" value="5.0" />
      <param name="max_angular_accel" value="5.0" />
      <param name="map_resolution" value="0.05" />
      <param name="linear_velocity_resolution" value="0.05" />
      <param name="angular_velocity_resolution" value="0.05" />
      <param name="robot_radius" value="0.1" />
      <param name="time_horizon" value="2.0" />
      <param name="k_v" value="10.0" />
      <param name="k_w" value="5.0" />

      <remap from="/obstacle_avoidance/scans" to="/scan"/>
      <remap from="/obstacle_avoidance/output_velocity" to="/cmd_vel_mux/input/navi"/>
      <!-- <remap from="/obstacle_avoidance/current_velocity" to="/vrep/twistStatus"/>
      <remap from="/obstacle_avoidance/output_velocity" to="/vrep/twistCommand"/> -->
  </node>

  <!-- <node pkg="topic_tools" type="mux" name="cmd_mux" 
      args="/obstacle_avoidance/command_velocity /teleop/twistCommand /mux/autoCommand">
  </node> -->
<!-- 
   <node pkg="topic_tools" type="mux" name="cmd_mux" 
      args="/obstacle_avoidance/command_velocity /teleop_velocity_smoother/raw_cmd_vel /cmd_vel_mux">
  </node> -->
<!-- 
  <node pkg="turtlebot_teleop" type="turtlebot_teleop_joy" name="turtlebot_teleop_joystick">
    <param name="scale_angular" value="1.5"/>
    <param name="scale_linear" value="0.5"/>
    <remap from="turtlebot_teleop_joystick/cmd_vel" to="/teleop_velocity_smoother/raw_cmd_vel"/>
  </node> -->
<!-- 
     <node pkg="joy" type="joy_node" name="joystick">
        <param name="autorepeat_rate" value="10"/>
  </node> -->


</launch>
